Example/SygicUIKit/GradientProgressExampleController.swift

```diff
@@ -0,0 +1,112 @@
+    
+    var gradients = [SYUICircleGradientProgressView]()
+    var timer: Timer?
+    var progress: Int = 0
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+        
+        let defaultGradient = SYUICircleGradientProgressView()
+        defaultGradient.frame = CGRect(x: 0, y: 100, width: 100, height: 100)
+        gradients.append(defaultGradient)
+        
+        let fullGradient = SYUICircleGradientProgressView()
+        fullGradient.dashLength = 1000
+        fullGradient.dashSpace = 0
+        fullGradient.frame = CGRect(x: 100, y: 100, width: 100, height: 100)
+        fullGradient.backgroundColor = .lightGray
+        gradients.append(fullGradient)
+        
+        let cutGradient = SYUICircleGradientProgressView()
+        cutGradient.progressOffset = 0.3
+        cutGradient.frame = CGRect(x: 200, y: 100, width: 100, height: 100)
+        gradients.append(cutGradient)
+        
+        let blueGradient = SYUICircleGradientProgressView()
+        blueGradient.gradientColors = [UIColor.white.cgColor, UIColor.blue.cgColor]
+        blueGradient.frame = CGRect(x: 0, y: 200, width: 100, height: 100)
+        gradients.append(blueGradient)
+        
+        let blueGradient2 = SYUICircleGradientProgressView()
+        blueGradient2.gradientColors = [UIColor.white.cgColor, UIColor.blue.cgColor]
+        blueGradient2.dashLength = 30
+        blueGradient2.frame = CGRect(x: 100, y: 200, width: 100, height: 100)
+        gradients.append(blueGradient2)
+        
+        let blueGradient3 = SYUICircleGradientProgressView()
+        blueGradient3.gradientColors = [UIColor.white.cgColor, UIColor.blue.cgColor]
+        blueGradient3.dashSpace = 10
+        blueGradient3.dashWidth = 10
+        blueGradient3.frame = CGRect(x: 200, y: 200, width: 100, height: 100)
+        gradients.append(blueGradient3)
+        
+        let bigGradient = SYUICircleGradientProgressView()
+        bigGradient.translatesAutoresizingMaskIntoConstraints = false
+        bigGradient.dashCap = .round
+        bigGradient.dashSpace = 25
+        bigGradient.dashWidth = 20
+        bigGradient.gradientColors = [UIColor.green.cgColor, UIColor.orange.cgColor, UIColor.red.cgColor]
+        bigGradient.progressOffset = 0.3
+        bigGradient.backgroundColor = .lightGray
+        gradients.append(bigGradient)
+        
+        for g in gradients {
+            view.addSubview(g)
+        }
+        
+        bigGradient.widthAnchor.constraint(equalToConstant: 300).isActive = true
+        bigGradient.heightAnchor.constraint(equalTo: bigGradient.widthAnchor, multiplier: 1).isActive = true
+        bigGradient.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        bigGradient.bottomAnchor.constraint(equalTo: view.safeBottomAnchor).isActive = true
+        
+        Timer.scheduledTimer(withTimeInterval: 1, repeats: true) { _ in
```

Comment:
Tuto si asi zabudol priradit tento timer do property `timer`, cize to teraz leakuje. Ale kazdopadne, nechcel si tu pouzit iba weakSelf? netreba si potom zbytocne drzat referenciu a pouzivat `viewWillDisappear`
```
guard let weakSelf = self else {
    timer.invalidate()
    return
}
```

Reply:
👍